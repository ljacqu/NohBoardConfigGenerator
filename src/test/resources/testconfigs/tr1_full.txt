[width = 40]
[height = 40]

Keys:
Inv Esc
Save F5
Load F6
↑ Up [marginLeft = 1.2]

LStp W
RStp E
Roll R
← Left [marginLeft = 0.2]
↓ Down
→ Right

Walk S
Jmp D
Actn F
Guns Space   [marginLeft = 0.2, width = 1.5]
Look NumPad0 [width = 61px]