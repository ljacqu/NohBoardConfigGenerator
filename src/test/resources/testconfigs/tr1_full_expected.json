{
  "Elements": [
    {
      "__type": "KeyboardKey",
      "Id": 1,
      "Boundaries": [
        {
          "X": 5,
          "Y": 5
        },
        {
          "X": 45,
          "Y": 5
        },
        {
          "X": 45,
          "Y": 45
        },
        {
          "X": 5,
          "Y": 45
        }
      ],
      "KeyCodes": [
        27
      ],
      "Text": "Inv",
      "TextPosition": {
        "X": 25,
        "Y": 25
      },
      "ChangeOnCaps": false,
      "ShiftText": "Inv"
    },
    {
      "__type": "KeyboardKey",
      "Id": 2,
      "Boundaries": [
        {
          "X": 46,
          "Y": 5
        },
        {
          "X": 86,
          "Y": 5
        },
        {
          "X": 86,
          "Y": 45
        },
        {
          "X": 46,
          "Y": 45
        }
      ],
      "KeyCodes": [
        116
      ],
      "Text": "Save",
      "TextPosition": {
        "X": 66,
        "Y": 25
      },
      "ChangeOnCaps": false,
      "ShiftText": "Save"
    },
    {
      "__type": "KeyboardKey",
      "Id": 3,
      "Boundaries": [
        {
          "X": 87,
          "Y": 5
        },
        {
          "X": 127,
          "Y": 5
        },
        {
          "X": 127,
          "Y": 45
        },
        {
          "X": 87,
          "Y": 45
        }
      ],
      "KeyCodes": [
        117
      ],
      "Text": "Load",
      "TextPosition": {
        "X": 107,
        "Y": 25
      },
      "ChangeOnCaps": false,
      "ShiftText": "Load"
    },
    {
      "__type": "KeyboardKey",
      "Id": 4,
      "Boundaries": [
        {
          "X": 176,
          "Y": 5
        },
        {
          "X": 216,
          "Y": 5
        },
        {
          "X": 216,
          "Y": 45
        },
        {
          "X": 176,
          "Y": 45
        }
      ],
      "KeyCodes": [
        38
      ],
      "Text": "↑",
      "TextPosition": {
        "X": 196,
        "Y": 25
      },
      "ChangeOnCaps": false,
      "ShiftText": "↑"
    },
    {
      "__type": "KeyboardKey",
      "Id": 5,
      "Boundaries": [
        {
          "X": 5,
          "Y": 46
        },
        {
          "X": 45,
          "Y": 46
        },
        {
          "X": 45,
          "Y": 86
        },
        {
          "X": 5,
          "Y": 86
        }
      ],
      "KeyCodes": [
        87
      ],
      "Text": "LStp",
      "TextPosition": {
        "X": 25,
        "Y": 66
      },
      "ChangeOnCaps": false,
      "ShiftText": "LStp"
    },
    {
      "__type": "KeyboardKey",
      "Id": 6,
      "Boundaries": [
        {
          "X": 46,
          "Y": 46
        },
        {
          "X": 86,
          "Y": 46
        },
        {
          "X": 86,
          "Y": 86
        },
        {
          "X": 46,
          "Y": 86
        }
      ],
      "KeyCodes": [
        69
      ],
      "Text": "RStp",
      "TextPosition": {
        "X": 66,
        "Y": 66
      },
      "ChangeOnCaps": false,
      "ShiftText": "RStp"
    },
    {
      "__type": "KeyboardKey",
      "Id": 7,
      "Boundaries": [
        {
          "X": 87,
          "Y": 46
        },
        {
          "X": 127,
          "Y": 46
        },
        {
          "X": 127,
          "Y": 86
        },
        {
          "X": 87,
          "Y": 86
        }
      ],
      "KeyCodes": [
        82
      ],
      "Text": "Roll",
      "TextPosition": {
        "X": 107,
        "Y": 66
      },
      "ChangeOnCaps": false,
      "ShiftText": "Roll"
    },
    {
      "__type": "KeyboardKey",
      "Id": 8,
      "Boundaries": [
        {
          "X": 136,
          "Y": 46
        },
        {
          "X": 176,
          "Y": 46
        },
        {
          "X": 176,
          "Y": 86
        },
        {
          "X": 136,
          "Y": 86
        }
      ],
      "KeyCodes": [
        37
      ],
      "Text": "←",
      "TextPosition": {
        "X": 156,
        "Y": 66
      },
      "ChangeOnCaps": false,
      "ShiftText": "←"
    },
    {
      "__type": "KeyboardKey",
      "Id": 9,
      "Boundaries": [
        {
          "X": 177,
          "Y": 46
        },
        {
          "X": 217,
          "Y": 46
        },
        {
          "X": 217,
          "Y": 86
        },
        {
          "X": 177,
          "Y": 86
        }
      ],
      "KeyCodes": [
        40
      ],
      "Text": "↓",
      "TextPosition": {
        "X": 197,
        "Y": 66
      },
      "ChangeOnCaps": false,
      "ShiftText": "↓"
    },
    {
      "__type": "KeyboardKey",
      "Id": 10,
      "Boundaries": [
        {
          "X": 218,
          "Y": 46
        },
        {
          "X": 258,
          "Y": 46
        },
        {
          "X": 258,
          "Y": 86
        },
        {
          "X": 218,
          "Y": 86
        }
      ],
      "KeyCodes": [
        39
      ],
      "Text": "→",
      "TextPosition": {
        "X": 238,
        "Y": 66
      },
      "ChangeOnCaps": false,
      "ShiftText": "→"
    },
    {
      "__type": "KeyboardKey",
      "Id": 11,
      "Boundaries": [
        {
          "X": 5,
          "Y": 87
        },
        {
          "X": 45,
          "Y": 87
        },
        {
          "X": 45,
          "Y": 127
        },
        {
          "X": 5,
          "Y": 127
        }
      ],
      "KeyCodes": [
        83
      ],
      "Text": "Walk",
      "TextPosition": {
        "X": 25,
        "Y": 107
      },
      "ChangeOnCaps": false,
      "ShiftText": "Walk"
    },
    {
      "__type": "KeyboardKey",
      "Id": 12,
      "Boundaries": [
        {
          "X": 46,
          "Y": 87
        },
        {
          "X": 86,
          "Y": 87
        },
        {
          "X": 86,
          "Y": 127
        },
        {
          "X": 46,
          "Y": 127
        }
      ],
      "KeyCodes": [
        68
      ],
      "Text": "Jmp",
      "TextPosition": {
        "X": 66,
        "Y": 107
      },
      "ChangeOnCaps": false,
      "ShiftText": "Jmp"
    },
    {
      "__type": "KeyboardKey",
      "Id": 13,
      "Boundaries": [
        {
          "X": 87,
          "Y": 87
        },
        {
          "X": 127,
          "Y": 87
        },
        {
          "X": 127,
          "Y": 127
        },
        {
          "X": 87,
          "Y": 127
        }
      ],
      "KeyCodes": [
        70
      ],
      "Text": "Actn",
      "TextPosition": {
        "X": 107,
        "Y": 107
      },
      "ChangeOnCaps": false,
      "ShiftText": "Actn"
    },
    {
      "__type": "KeyboardKey",
      "Id": 14,
      "Boundaries": [
        {
          "X": 136,
          "Y": 87
        },
        {
          "X": 196,
          "Y": 87
        },
        {
          "X": 196,
          "Y": 127
        },
        {
          "X": 136,
          "Y": 127
        }
      ],
      "KeyCodes": [
        32
      ],
      "Text": "Guns",
      "TextPosition": {
        "X": 166,
        "Y": 107
      },
      "ChangeOnCaps": false,
      "ShiftText": "Guns"
    },
    {
      "__type": "KeyboardKey",
      "Id": 15,
      "Boundaries": [
        {
          "X": 197,
          "Y": 87
        },
        {
          "X": 257,
          "Y": 87
        },
        {
          "X": 257,
          "Y": 127
        },
        {
          "X": 197,
          "Y": 127
        }
      ],
      "KeyCodes": [
        96
      ],
      "Text": "Look",
      "TextPosition": {
        "X": 227,
        "Y": 107
      },
      "ChangeOnCaps": false,
      "ShiftText": "Look"
    }
  ],
  "Height": 132,
  "Width": 263,
  "Version": 2
}